_search-plugins/ubi/schemas.md

```diff
@@ -0,0 +1,228 @@
+
+- `message` (size 1024) : An optional text message for the log entry. For example, for a `message_type` of `QUERY`, the `message` can contain the text related to a user's search.
+
+ <p id="event_attributes"> </p>
+
+- `event_attributes`: An extensible structure that describes any important context about the event. This structure consists of two primary structures: `position` and `object`. The structure is extensible so you can add custom information about the event, such as the event's timing, user, or session.
+  
+ Because the `ubi_events` index is configured to perform dynamic mapping, the index can become bloated with many new fields.
+  {: .warning} 
+
+ - `event_attributes.position`: A structure that contains information about the location of the event origin, such as screen x, y coordinates or the object's position in the result list:
+ 
+   - `event_attributes.position.ordinal`: Tracks the position within a list that a user could select (for example, selecting the 3rd element can be described as `event{onClick, results[4]}`).
+
+    - `event_attributes.position.{x,y}`: Tracks x and y values defined by the client.
+
+    - `event_attributes.position.page_depth`: Tracks the page depth of the results.
+
+    - `event_attributes.position.scroll_depth`: Tracks the scroll depth of the page results.
+
+    - `event_attributes.position.trail`: A text field for tracking the path/trail that a user took to get to this location.
+  
+  - `event_attributes.object`: Contains identifying information about the object returned from the query that the user interacts with (for example, a book, product, or post).
+   The `object` structure can refer to the object by internal ID or object ID. The`object_id` is the id that links prior queries to this object. This field comprises the following subfields:
+  
+    - `event_attributes.object.internal_id`: A unique ID that OpenSearch can use to internally to index the object, for example, the `_id` field in the indexes.
+
+      <p id="object_id">
+
+    - `event_attributes.object.object_id`:  An ID by which a user can find the object instance within the **document corpus**. Examples include `ssn`, `isbn`, or `ean`.  Variants need to be incorporated in the `object_id`, so a red t-shirt's `object_id` should be its SKU.
+      Initializing UBI requires mapping the document index's primary key to this `object_id`.
+      </p>
+
+      <p id="object_id_field">
+         
+    - `event_attributes.object.object_id_field`: Indicates the type/class of the object and the name of the search index field that contains the `object_id`.  
+
+    - `event_attributes.object.description`: An optional description of the object.
+
+    - `event_attributes.object.object_detail`: Optional text for specifying further data about the object.
+          
+    - *extensible fields*: Any new indexed fields in the `object` dynamically expands this schema.
```

Comment:
Why is this called out as a warning? Would be better to keep it as a regular bullet then add a warning underneath it.

Reply:
I wasn't sure how to be most emphatic about this point.  If a company were to broadly adopt UBI across multiple applications and the developers do not take heed in this warning, the index could grow to have hundreds or thousands of sparsely populated fields.